octopoes/models/ooi/web.py

```diff
@@ -194,3 +197,95 @@ def format_reference_human_readable(cls, reference: Reference) -> str:
+
+
+class HTTPCookie(OOI):
+    # https://datatracker.ietf.org/doc/html/rfc6265
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies
+    object_type: Literal["HTTPCookie"] = "HTTPCookie"
+
+    _natural_key_attrs = [
+        "name",
+        "domain",
+        "path",
+    ]  # does not include network, as there is not concept of a network only hostnames in the browsers cookie handling.
+
+    # httpheader: Reference(ReferenceField(HTTPHeaderURL, max_inherit_scan_level=4))  # ideally we should keep track of which header was parsed into this cookie object
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3
+    name: str
+    value: str
+    expirytime: Optional[datetime]
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p6
+    domain: Reference = ReferenceField(Hostname, max_inherit_scan_level=4)
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p7
+    path: str = "/"
+    creationtime: datetime
+    # last-access-time: # not used in openkat context
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p3
+    persistent: bool = False
+
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p6
+    hostonly: bool = False
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p8
+    secureonly: bool = False
+    # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p9
+    httponly: bool = False
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite
+    samesite: bool = False
+    maxage: Optional[int]
+
+    @classmethod
+    def format_reference_human_readable(cls, reference: Reference) -> str:
+        tokenized = reference.tokenized
+        return f"{tokenized.name} @ {tokenized.domain}"
+
+    @classmethod
+    def fromstring(cls, responsedomain, cookie) -> Iterator[OOI]:
+        now = datetime.now(timezone.utc)
+        # https://docs.python.org/3/library/http.cookies.html
+        try:
+            parsedcookie = SimpleCookie(cookie)
+        except CookieError as cookieerror:
+            yield Finding(cookieerror)
+        for name, morsel in parsedcookie.items():
+            # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p6
+            hostonly = False
+            if not morsel.domain:
+                hostonly = True
+                morsel.domain = responsedomain
+
+            # https://datatracker.ietf.org/doc/html/rfc6265#section-5.3 p3
+            persistent = False
+            expires = float("inf")
+            if "max-age" in morsel:
+                persistent = True
+                try:
+                    maxage = min(
+                        1e8, int(morsel["max-age"])
+                    )  # limit to make sure we dont trip over calculating a date millions of years in the future
+                except ValueError:
+                    persistent = False
+                else:
+                    expires = now + timedelta(maxage)
```

Comment:
According to https://datatracker.ietf.org/doc/html/rfc6265#section-5.2.2 the max-age attribute is in seconds, while in this expression you're adding it as days